features/device-config/rest/src/main/java/org/opennms/features/deviceconfig/rest/impl/DefaultDeviceConfigRestService.java

```diff
@@ -102,24 +135,188 @@ public Response getDeviceConfigs(
         }
     }
 
+    /** Get DeviceConfig-related OnmsMonitoredServices for the given ip interface ids. */
+    private List<OnmsMonitoredService> getDeviceConfigServices(List<Integer> ipInterfaceIds) {
+        List<OnmsMonitoredService> services = monitoredServiceDao.findByServiceTypeAndIpInterfaceId(DEVICE_CONFIG_SERVICE_PREFIX, ipInterfaceIds);
+
+        return services;
+    }
+
+    /**
+     * Find corresponding service for each DTO (based on IP address and config type) and
+     * update DTO schedule information based on service metadata.
+     */
+    private void populateScheduleInfo(List<DeviceConfigDTO> dtos, Map<Integer, List<OnmsMonitoredService>> serviceMap) {
+        final Date currentDate = new Date();
+
+        for (DeviceConfigDTO dto : dtos) {
+            if (serviceMap.containsKey((Integer) dto.getIpInterfaceId())) {
+                final List<OnmsMonitoredService> serviceList = serviceMap.get((Integer) dto.getIpInterfaceId());
+
+                final String configType = Strings.isNullOrEmpty(dto.getConfigType()) ? ConfigType.Default : dto.getConfigType();
+                final String serviceName = DEVICE_CONFIG_SERVICE_PREFIX + "-" + configType;
+
+                final Optional<String> scheduleValue = getScheduleValue(serviceList, serviceName);
+
+                if (scheduleValue.isPresent()) {
+                    final ScheduleInfo scheduleInfo = getScheduleInfo(currentDate, scheduleValue.get());
+
+                    dto.setNextScheduledBackupDate(scheduleInfo.getNextScheduledBackup());
+                    dto.setScheduledInterval(scheduleInfo.getScheduleInterval());
+                }
+            }
+        }
+    }
+
+    public static Optional<String> getScheduleValue(final List<OnmsMonitoredService> serviceList, final String serviceName) {
+        final var matchingService =
+            serviceList.stream()
+            .filter(x -> x.getServiceName().equalsIgnoreCase(serviceName))
+            .findFirst();
+
+        if (matchingService.isPresent()) {
+            return getScheduleMetaDataValue(matchingService.get().getMetaData());
+        }
+
+        return Optional.empty();
```

Comment:
```suggestion
        return matchingService.flatMap(service -> getScheduleMetaDataValue(service.getMetaData());
```

Reply:
I will make this change in subsequent PR